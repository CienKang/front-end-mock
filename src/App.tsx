import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import './App.css';
import { Header } from './components';
import { ErrorPage, MainPage } from './pages';

function App() {
    return (
        <div className='all'>
            <Header />
            <BrowserRouter >
                <Routes >
                    <Route path='/' element={<MainPage />} />
                    <Route path='/error' element={<ErrorPage />} />
                </Routes>
            </BrowserRouter>
        </div>
    );
}

export default App;