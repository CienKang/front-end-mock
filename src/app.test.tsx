import React from 'react';
import { render,screen } from '@testing-library/react';
import App from './App';

describe('App', () => {

    it('should render the component correctly and show header text when launched',()=>{
        render(<App/>);
        
        expect(screen.getByRole('banner')).toBeTruthy();
        expect(screen.getByText(/my shelf/i)).toBeTruthy();
        expect(screen.getByText(/record/i)).toBeTruthy();
    });

    it('should match with the previous snapshots',()=>{
        const {asFragment} = render(<App/>);
        expect(asFragment()).toMatchSnapshot();
    });

});