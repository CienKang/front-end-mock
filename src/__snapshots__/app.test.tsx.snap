// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match with the previous snapshots 1`] = `
<DocumentFragment>
  <div
    class="all"
  >
    <header>
      <span>
        My 
        <b
          class="bolder"
        >
          Record 
        </b>
         Shelf
      </span>
    </header>
    <div>
      <p>
        loading...
      </p>
    </div>
  </div>
</DocumentFragment>
`;
